scipy/signal/_fir_filter_design.py

```diff
@@ -426,26 +432,27 @@ def firwin(numtaps, cutoff, *, width=None, window='hamming', pass_zero=True,
                              f'{pass_zero}')
     pass_zero = bool(operator.index(pass_zero))  # ensure bool-like
 
-    pass_nyquist = bool(cutoff.size & 1) ^ pass_zero
+    pass_nyquist = bool(xp_size(cutoff) & 1) ^ pass_zero
```

Comment:
Another matter of taste comment: I think
```suggestion
    pass_nyquist = (xp_size(cutoff) % 2 == 0) != pass_zero
```
is easier to comprehend (also due to ` numtaps % 2 == 0` being used in the line below).

Reply:
Would certainly be more readable! However, should it be `== pass_zero` though?

```
In [11]: for x in (42, 43):
    ...:     for pass_zero in (True, False):
    ...:         old = bool(x & 1) ^ pass_zero
    ...:         new = (x % 2 == 0) != pass_zero
    ...:         print(x, pass_zero, old, new)
    ...: 
42 True True False
42 False False True
43 True False True
43 False True False
```

Reply:
Of course :clown_face:! Fixed [here](https://github.com/ev-br/scipy/pull/22).

Reply:
Thanks! Included in https://github.com/scipy/scipy/pull/22343/commits/6c30b7d0ef8200c39c497b01a40eafe21d358139